pkg/fetch/fetch.go

```diff
@@ -17,18 +17,70 @@
 package fetch
 
 import (
+	"fmt"
+	"net/url"
+	"path"
+	"path/filepath"
 	"time"
 
 	"github.com/prometheus/client_golang/prometheus"
 )
 
+// realPath replaces path root of p by r if r != "" and returns p otherwise.
```

Comment:
The functions added here are used by the fetchers.
Please look through their implementations - that should fix your concerns in https://github.com/TheThingsNetwork/lorawan-stack/pull/1341#issuecomment-529835157
The algorithm implemented is defined in https://github.com/TheThingsNetwork/lorawan-stack/issues/1340